File: cross-application-feature-development-management/Directories/HostingDirectory/FeatureDirectory/AutomationsDirectory/EnvironmentVariablesTemplateFiles/DirectoriesMultitudeServingOrderRectoActionOpen.cs
using System.Text;
using cross_application_feature_development_management.Helpers;
using cross_application_feature_development_management.Names;

namespace cross_application_feature_development_management.Directories.HostingDirectory.FeatureDirectory.AutomationsDirectory.EnvironmentVariablesTemplateFiles
{
    public class DirectoriesMultitudeServingOrderRectoActionOpen
    (
        StringHelpers stringHelpers,
        PrimaryApplication primaryApplication,
        SecondaryApplication secondaryApplication,
        FeatureName featureName
    )
    {
        public Dictionary<string, string> PairUpVariablesWithTheirValue(
            string fileNamePath,
            Dictionary<string, string> environmentVariablesSourceDictionary
        )
        {
            Dictionary<string, string> fileContentDictionaryToWriteToFile = [];

            const int bufferSize = 128;
            using var fileStream = File.OpenRead(fileNamePath);
            using var streamReader = new StreamReader(fileStream, Encoding.UTF8, true, bufferSize);

            while (streamReader.ReadLine() is { } line)
            {
                var brokenLine = line.Split("=");
                var key = brokenLine[0];
                var value = brokenLine[1];
                
                if(environmentVariablesSourceDictionary.TryGetValue(key, out var val)){
                    switch (key)
                    {
                        case "COMMAND":
                        {
                            var wrappedVal = stringHelpers.WrapInQuotationMarks("open");
                            fileContentDictionaryToWriteToFile.Add(key, wrappedVal ?? "");
                            break;
                        }
                        case "APPLICATION":
                        {
                            var wrappedVal = stringHelpers.WrapInQuotationMarks("directory-management");
                            fileContentDictionaryToWriteToFile.Add(key, wrappedVal ?? "");
                            break;
                        }
                        case "ORDER":
                        {
                            var wrappedVal = stringHelpers.WrapInQuotationMarks("reverse");
                            fileContentDictionaryToWriteToFile.Add(key, wrappedVal ?? "");
                            break;
                        }
                        case "STARTUP_DIRECTORY_LOCATION":
                        {
                            environmentVariablesSourceDictionary.TryGetValue(key, out var val1);
                            fileContentDictionaryToWriteToFile.Add(key, val1 ?? "");
                            break;
                        }
                    case "CAFDEM_EXECUTIVE_FILE_ADDRESS_CONTAINING_DIRECTORY":
                        {
                            if(environmentVariablesSourceDictionary.TryGetValue(
                                "CAFDEM_EXECUTIVE_FILE_ADDRESS",
                                out var cafdemExecutiveFileAddress))
                                {
                                    var striped = stringHelpers.StripQuotationMarks(cafdemExecutiveFileAddress);
                                    var dirName = Path.GetDirectoryName(striped);
                                    fileContentDictionaryToWriteToFile.Add(key, dirName ?? "");
                                }
                            break;
                        }
                    case "PRIMARY_APPLICATION_NAME":
                        {
                            var val3 = primaryApplication.GetName();
                            var wrappedVal = stringHelpers.WrapInQuotationMarks(val3);
                            fileContentDictionaryToWriteToFile.Add(key, wrappedVal ?? "");
                            break;
                        }
                    case "FEATURE_NAME":
                        {
                            var val2 = featureName.GetName();
                            var wrappedVal = stringHelpers.WrapInQuotationMarks(val2);
                            fileContentDictionaryToWriteToFile.Add(key, wrappedVal ?? "");
                            break;
                        }
                    case "SECONDARY_APPLICATION_NAME":
                        {
                            var val3 = secondaryApplication.GetName();
                            var wrappedVal = stringHelpers.WrapInQuotationMarks(val3);
                            fileContentDictionaryToWriteToFile.Add(key, wrappedVal ?? "");
                            break;
                        }
                        default:
                        {
                            environmentVariablesSourceDictionary.TryGetValue(key, out var val2);
                            fileContentDictionaryToWriteToFile.Add(key, val2 ?? "");
                            break;
                        }
                    }
                }
            }
            return fileContentDictionaryToWriteToFile;
        }
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/Directories/Hosting/Feature/Automations/EnvironmentVariablesTemplateFiles/DirectoriesMultitude/DirectoriesMultitudeServingOrderRectoActionOpen.fs
namespace cafdemwimu.console.Directories.Hosting.Feature.Automations.EnvironmentVariablesTemplateFiles.DirectoriesMultitude

open System.IO
open System.Text
open System.Collections.Generic
open Microsoft.Extensions.Logging
open cafdemwimu.console.Applications.Cafdemalihapa
open cafdemwimu.console.Directories.NullHelpers
open cafdemwimu.console.Directories.Hosting
open cafdemwimu.console.Directories.Repository
open cafdemwimu.console.Helpers
open cafdemwimu.console.Names

type DirectoriesMultitudeServingOrderRectoActionOpen
    (
        stringHelpers: StringHelpers,
        primaryApplication: PrimaryApplication,
        secondaryApplication: SecondaryApplication,
        featureName: FeatureName,
        repositoryDirectory: RepositoryDirectory,
        codeBase: CodeBase,
        logger: ILogger<DirectoriesMultitudeServingOrderRectoActionOpen>
    ) =
    member _.PairUpVariablesWithTheirValue(fileNamePath: string, environmentVariablesSourceDictionary: Dictionary<string, string>) =
        let fileContentDictionaryToWriteToFile = Dictionary<string, string>()

        let bufferSize = 128
        use fileStream = File.OpenRead(fileNamePath)
        use streamReader = new StreamReader(fileStream, Encoding.UTF8, true, bufferSize)

        let mutable line = streamReader.ReadLine()
        while not (isNull line) do
            let brokenLine = line.Split('=')
            let key = brokenLine.[0]
            let _value = brokenLine.[1]
            match environmentVariablesSourceDictionary.TryGetValue key with
            | true, _ ->
                try
                    match key with
                    | "COMMAND" -> fileContentDictionaryToWriteToFile.Add(key, stringHelpers.WrapInQuotationMarks("open"))
                    | "APPLICATION" -> fileContentDictionaryToWriteToFile.Add(key, stringHelpers.WrapInQuotationMarks("directory-management"))
                    | "ORDER" -> fileContentDictionaryToWriteToFile.Add(key, stringHelpers.WrapInQuotationMarks("reverse"))
                    | "FORMAT" -> fileContentDictionaryToWriteToFile.Add(key, stringHelpers.WrapInQuotationMarks("json"))
                    | "FILEMENT" -> fileContentDictionaryToWriteToFile.Add(key, stringHelpers.WrapInQuotationMarks("split"))
                    | "REPOSITORY_DIRECTORY" -> fileContentDictionaryToWriteToFile.Add(key, stringHelpers.WrapInQuotationMarks(repositoryDirectory.GetPath()))
                    | "CAFDEM_EXECUTIVE_FILE_ADDRESS_CONTAINING_DIRECTORY" ->
                        match environmentVariablesSourceDictionary.TryGetValue("CAFDEM_EXECUTIVE_FILE_ADDRESS") with
                        | true, cafdemExecutiveFileAddress ->
                            let striped = stringHelpers.StripQuotationMarks(cafdemExecutiveFileAddress)
                            let dirName = Path.GetDirectoryName(striped: string)
                            fileContentDictionaryToWriteToFile.Add(key, orEmpty dirName)
                        | _ -> ()
                    | "PRIMARY_APPLICATION_NAME" -> fileContentDictionaryToWriteToFile.Add(key, stringHelpers.WrapInQuotationMarks(primaryApplication.GetName()))
                    | "FEATURE_NAME" -> fileContentDictionaryToWriteToFile.Add(key, stringHelpers.WrapInQuotationMarks(featureName.GetName()))
                    | "CODE_BASE" -> fileContentDictionaryToWriteToFile.Add(key, stringHelpers.WrapInQuotationMarks(codeBase.GetCodeBaseValue()))
                    | "SECONDARY_APPLICATION_NAME" -> fileContentDictionaryToWriteToFile.Add(key, stringHelpers.WrapInQuotationMarks(secondaryApplication.GetName()))
                    | "HOSTING_DIRECTORY" -> fileContentDictionaryToWriteToFile.Add(key, stringHelpers.WrapInQuotationMarks(HostingDirectory.GetPath()))
                    | _ ->
                        let _, val2 = environmentVariablesSourceDictionary.TryGetValue key
                        fileContentDictionaryToWriteToFile.Add(key, orEmpty val2)
                with _ ->
                    logger.LogError("DirectoriesMultitudeServingOrderRectoActionOpen: the key could not be processed: {Key}", key)
            | _ -> ()
            line <- streamReader.ReadLine()

        fileContentDictionaryToWriteToFile
